charts/internal/shoot-system-components/charts/rook-ceph/templates/toolbox.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: rook-ceph-tools
+  namespace: rook-ceph # namespace:cluster
+  labels:
+    app: rook-ceph-tools
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: rook-ceph-tools
+  template:
+    metadata:
+      labels:
+        app: rook-ceph-tools
+    spec:
+      dnsPolicy: ClusterFirstWithHostNet
+      containers:
+      - name: rook-ceph-tools
+        image: rook/ceph:master
+        command: ["/tini"]
+        args: ["-g", "--", "/usr/local/bin/toolbox.sh"]
+        imagePullPolicy: IfNotPresent
+        env:
+          - name: ROOK_CEPH_USERNAME
+            valueFrom:
+              secretKeyRef:
+                name: rook-ceph-mon
+                key: ceph-username
+          - name: ROOK_CEPH_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: rook-ceph-mon
+                key: ceph-secret
+        volumeMounts:
+          - mountPath: /etc/ceph
+            name: ceph-config
+          - name: mon-endpoint-volume
```

Comment:
Have a checksum annotation for this mounted volume, see e.g. https://github.com/gardener/gardener-extension-provider-aws/blob/e300a1678cf5325e845ecf56bb43a66ddfe29adb/charts/gardener-extension-admission-aws/charts/runtime/templates/deployment.yaml#L19

Reply:
Checksum of what? The example you linked to is of a single file, this is a directory with dynamic data.

Reply:
Hm, okay, who creates the `rook-ceph-mon-endpoints` config map? I don't see it here in the Helm chart, so I assume it's created dynamically? Though, how does this work; wouldn't this pod fail to start in a fresh cluster because the config map is missing?

Reply:
Not actually sure. It has been a few months, I may need to go back to the original source of this.

BTW, does Gardener have the ability to just load a helm chart dynamically from a registry? There are official rook-ceph helm charts, which would make this much easier.

Reply:
AFAIK the mon config is created by the rook operator when a ceph cluster is deployed. It's right that the toolbox is not started before the cluster is ready. But the toolbox also makes no sense without a ceph cluster. FWIW the general idea is that you start the toolbox on demand and otherwise it's not even present on the cluster.

Reply:
Yeah, I don't know much about that part, it was mostly copied.

What do you suggest @christianhuening, that we remove it?

Reply:
i'd leave it as it is. The toolbox will be pending until the cluster is up, but that's fine.